lib/modules/datasource/docker/index.ts

```diff
@@ -824,42 +824,113 @@ export class DockerDatasource extends Datasource {
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
-      let manifestResponse = await this.getManifestResponse(
-        registryHost,
-        dockerRepository,
-        newTag,
-        'head'
-      );
-      if (manifestResponse) {
-        if (hasKey('docker-content-digest', manifestResponse.headers)) {
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
+        const manifestResponse = await this.getManifestResponse(
```

Comment:
This will count to rate limits on docker hub. So i think we need to add our cache decorator to `getManifestResponse` to reuse the current response.

Reply:
Now without the HEAD request for architecture digests, the total number of requests to Docker Hub will be the same as for the existing implementation, when assuming the HEAD request counts too.

It seems, the cache decorator is only used with public methods so far but not `getConfigDigest()`, which potentially may face the same issues?! I would suspect that the cache decorator for `getDigest()` is already ensuring the digest is reused, where possible. I'm not familiar with the cache implementation though.

Reply:
`HEAD` doesn't count, that's why i previously refactored to use `HEAD` requests.

Can we maybe do a `HEAD` request and check response `content-type`. If it's a image, don't fetch architecture.
Or better only first fetch both old and new digest and content-type via `HEAD`. if new content-type is a image-list and old content-type is a image -> fetch the architectures and use right digest.

We should at least do long term caching for blobs, as they should never change. This can be done in a separate PR.

Reply:
Validated via curl. we can use the content-type header

```sh
> curl -s -I -H 'autorization: Bearer xxxx' -H 'accept: application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json' https://index.docker.io/v2/library/node/manifests/latest
HTTP/2 200
content-type: application/vnd.docker.distribution.manifest.list.v2+json
date: Fri, 15 Jul 2022 09:31:57 GMT
docker-content-digest: sha256:c9504e6bdd0498b99acbf392b94e4d0d56b2c6a37e52b5766cfb909894d9f389
docker-distribution-api-version: registry/2.0
etag: "sha256:c9504e6bdd0498b99acbf392b94e4d0d56b2c6a37e52b5766cfb909894d9f389"
x-content-type-options: nosniff
content-length: 1206

> curl -s -I -H 'autorization: Bearer xxxx' -H 'accept: application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json' https://index.docker.io/v2/renovate/renovate/manifests/latest
HTTP/2 200
content-type: application/vnd.docker.distribution.manifest.v2+json
date: Fri, 15 Jul 2022 09:32:31 GMT
docker-content-digest: sha256:47545b2a0171fb65b98b1fe5aa819dc7a5999fee9cce696b706eb7deb266ecd1
docker-distribution-api-version: registry/2.0
etag: "sha256:47545b2a0171fb65b98b1fe5aa819dc7a5999fee9cce696b706eb7deb266ecd1"
x-content-type-options: nosniff
content-length: 8115


> curl -s -I -H 'autorization: Bearer xxxx' -H 'accept: application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json' https://index.docker.io/v2/renovate/renovate/manifests/sha256:47545b2a0171fb65b98b1fe5aa819dc7a5999fee9cce696b706eb7deb266ecd1
HTTP/2 200
content-type: application/vnd.docker.distribution.manifest.v2+json
date: Fri, 15 Jul 2022 09:32:31 GMT
docker-content-digest: sha256:47545b2a0171fb65b98b1fe5aa819dc7a5999fee9cce696b706eb7deb266ecd1
docker-distribution-api-version: registry/2.0
etag: "sha256:47545b2a0171fb65b98b1fe5aa819dc7a5999fee9cce696b706eb7deb266ecd1"
x-content-type-options: nosniff
content-length: 8115
```